plugin/src/main/kotlin/com/trevjonez/composer/ComposerTask.kt

```diff
@@ -88,9 +85,9 @@ abstract class ComposerTask : JavaExec(), ComposerConfigurator, ComposerTaskDsl
     val buildDirectory = project.layout.buildDirectory
     outputDir.convention(buildDirectory.dir(ComposerConfig.DEFAULT_OUTPUT_DIR))
     workDir.convention(buildDirectory.dir(ComposerConfig.DEFAULT_WORK_DIR))
-    setWorkingDir(workDir.map { it.also { it.asFile.mkdirs() } })
```

Comment:
This line errors on Gradle 7. Quote from user guide

> **Querying the mapped value of a provider before the producer has completed is now an error**
>
> Gradle 6.x warns users about the wrong behavior and then returns a possibly incorrect provider
value. Starting with 7.0 the same case will produce an error. Plugins and build scripts should be
adjusted to query the mapped value of a provider, for example a task output property, after the task
has completed.

Resolving output property before task finishes is now an error. I'm not 100% sure, but it looks like gradle will mkdir folder/file properties automatically.